src/tsolvers/UFLATHandler.cc

```diff
@@ -81,34 +74,28 @@ vec<PTRef> UFLATHandler::getSplitClauses() {
         arraySolver->getNewSplits(res);
         vec<PTRef> arrayLemmas;
         res.copyTo(arrayLemmas);
-        std::unordered_set<PTRef, PTRefHash> equalitiesWithAddedClauses;
         // HACK: If array solver needs to decide equality on interface vars, we need to add corresponding lemmas already here
         for (PTRef lemma : arrayLemmas) {
             if (logic.isOr(lemma)) {
                 for (PTRef lit : logic.getPterm(lemma)) {
                     PTRef atom = logic.isNot(lit) ? logic.getPterm(lit)[0] : lit;
-                    if (logic.isNumEq(atom) and knownEqualities.find(atom) == knownEqualities.end() and
-                        equalitiesWithAddedClauses.find(atom) == equalitiesWithAddedClauses.end()) {
+                    if (logic.isNumEq(atom) and equalitiesWithAddedInterfaceClauses.find(atom) == equalitiesWithAddedInterfaceClauses.end()) {
                         PTRef lhs = logic.getPterm(atom)[0];
                         PTRef rhs = logic.getPterm(atom)[1];
-                        if (std::find(interfaceVars.begin(), interfaceVars.end(), lhs) != interfaceVars.end() and
-                            std::find(interfaceVars.begin(), interfaceVars.end(), rhs) != interfaceVars.end()
-                        ) {
+                        if ((logic.isVar(lhs) or logic.isNumConst(lhs)) and (logic.isVar(rhs) or logic.isNumConst(rhs))) {
```

Comment:
```suggestion
                        if (isInterfaceVar(lhs) and isInterfaceVar(rhs)) {
```
using lambda
```
    auto isInterfaceVar = [this](PTRef arg){
        return logic.isVar(arg) or logic.isNumConst(arg);
    };
```
(or use another name)

Reply:
Changed!